uktrade/great-cms
# Add feedback button experiment to all pages

This PR adds a 'floating' feedback button to the core and domestic base templates. The button will appear on every domestic page, aside from the microsite pages, when:
- A user's screen-width is above 640px
- The feature flag `FEATURE_FEEDBACK_BUTTON_EXPERIMENT` is set to true
- Javascript is enabled in the user's browser

When a user clicks 'Give feedback' they will be taken to the existing /search/feedback form page, with the page they came from in the `?next=...` query parameter. Once they have submitted the form they will be taken to a new confirmation page, from which they can continue to the page they came on.

**To note**
- The PR also includes small tweaks to the styling of the feedback form to standardise it against other forms

**To test**
- Add `FEATURE_FEEDBACK_BUTTON_EXPERIMENT=true` to your secrets_do_not_commit file
- Go to any page, the button should appear in the bottom right when you have javascript enabled, and not appear if disabled
- After clicking the 'No thanks' link, the box should collapse again
- After clicking 'Give feedback' you should be taken to the feedback form, where the button should NOT appear
- After submitting you should be taken to `/search/feedback/success?next=/`, where the button should also NOT appear
- When you click 'Continue' you should be taken back to the page you originally came from

**Screenshots**
![Screenshot 2023-08-24 at 10 05 43](https://github.com/uktrade/great-cms/assets/22460823/479c6207-c1dc-4756-8cdb-8f4d4e9f0217)

![Screenshot 2023-08-24 at 10 05 56](https://github.com/uktrade/great-cms/assets/22460823/d22a2fb1-6025-4c29-ab8c-d9ec468f2875)

![Screenshot 2023-08-24 at 10 06 13](https://github.com/uktrade/great-cms/assets/22460823/dcf4659a-6283-4349-a31d-b359ec726948)


![Screenshot 2023-08-24 at 10 07 15](https://github.com/uktrade/great-cms/assets/22460823/0faf0007-3d09-43e9-b379-d27bdb6bae78)


### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/KLS-1075

### Reviewing help


### Housekeeping



### Merging
